src/State/Pagination/TraversablePartialPaginator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\State\Pagination;
+
+/**
+ * @template T of object
+ *
+ * @implements PaginatorInterface<T>
+ * @implements \IteratorAggregate<T>
+ */
+final class TraversablePartialPaginator implements \IteratorAggregate, PartialPaginatorInterface
+{
+    /**
+     * @param \Traversable<T> $traversable
+     */
+    public function __construct(private readonly \Traversable $traversable, private readonly float $currentPage, private readonly float $itemsPerPage)
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCurrentPage(): float
+    {
+        return $this->currentPage;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItemsPerPage(): float
+    {
+        return $this->itemsPerPage;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count(): int
+    {
+        return iterator_count($this->getIterator());
+    }
```

Comment:
just wondering what we should count here :thinking: 

I had a problem of `Cannot traverse an already closed generator` when using this, and passed the `$count` as a constructor property. WDYT?